src/ResourceManager/Advisor/Commands.Advisor/Cmdlets/Utilities/RecommendationHelper.cs

```diff
@@ -0,0 +1,196 @@
+                    return recommendationEntry;
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Filter recommendations by given recommendationNameList.
+        /// </summary>
+        /// <param name="recListTobeFiltered">List to be filtered</param>
+        /// <param name="recommendationNameList">List of recommendation-names, type GUID.</param>
+        /// <returns>Recommendation list of PsAzureAdvisorResourceRecommendationBase type</returns>
+        public static List<PsAzureAdvisorResourceRecommendationBase> RecomendationFilterByRecommendation(IEnumerable<PsAzureAdvisorResourceRecommendationBase> recListTobeFiltered, List<string> recommendationNameList)
+        {
+            List<PsAzureAdvisorResourceRecommendationBase> returnList = new List<PsAzureAdvisorResourceRecommendationBase>();
+            foreach (string recommendaitonName in recommendationNameList)
+            {
+                returnList.Add(RecomendationFilterByRecommendation(recListTobeFiltered, recommendaitonName));
+            }
+
+            return returnList;
+        }
+
+        /// <summary>
+        /// Parse the subscriptionId from the resourceId.
+        /// </summary>
+        /// <param name="resourceID">ResourceId of recommendation</param>
+        /// <returns>SubscriptionId as string</returns>
+        public static string GetSubscriptionIdfromResoureID(string resourceID)
+        {
+            string subscriptionID = string.Empty;
+
+            if (resourceID.Contains("subscriptions/") && !resourceID.Contains("resourceGroups/"))
+            {
+                int startIndex = resourceID.IndexOf("configurations/") + 16;
+
+                subscriptionID = resourceID.Substring(startIndex);
+            }
+
+            return subscriptionID;
+        }
+
+        /// <summary>
+        /// Parse the subscriptionId from the resourceId.
+        /// </summary>
+        /// <param name="resourceID">ResourceId of recommendation</param>
+        /// <returns>SubscriptionId as string</returns>
+        public static string GetResourceGroupfromResoureID(string resourceID)
+        {
+            string resourceGroup = string.Empty;
+
+            if (resourceID.Contains("resourceGroups/"))
+            {
+                int startIndex = resourceID.IndexOf("resourceGroups/") + 15;
+                int endIndex = resourceID.IndexOf("providers/Microsoft.Advisor") - 2;
+
+                int cutLength = endIndex - startIndex + 1;
+                resourceGroup = resourceID.Substring(startIndex, cutLength);
+            }
+
+            return resourceGroup;
+        }
+
+        /// <summary>
+        /// Parse the subscriptionId from the resourceId.
+        /// </summary>
+        /// <param name="resourceID">ResourceId of recommendation</param>
+        /// <returns>SubscriptionId as string</returns>
+        public static string GetFullResourceUrifromResoureID(string resourceID)
```

Comment:
Please check your spelling across the board.  This should be *GetFullResourceUriFromResourceID